### ISSUE TITLE
[Bug Report] `loadbalancer` submodule's `zones` variable does not allow setting value to explicit `null` to disable zonal deployment

### ISSUE THREAD

**Issue by jinkang23** [PaloAltoNetworks/terraform-azurerm-swfw-modules#59]
### Describe the bug

Certain Azure regions such as `westus` does not support Availability Zones, therefore, requires setting the value of `var.zones` to `null`. Description of this variable suggest explicitly settting this variable to `null` to disable zonal deployment, however, it cannot be set to `null` because the variable is set to `nullable = false`. 

### Module Version

v3.0.0

### Terraform version

1.8.5

### Expected behavior

Changing `nullable = true` allows setting this value of this variable to `null` which in turn passes that value to both `azurerm_public_ip` resource and `frontend_ip_configuration` of `azurerm_lb` resource to disable zonal deployment.  

Example of change to the `zones` variable:  

```hcl
variable "zones" {
  description = <<-EOF
  Controls zones for Load Balancer's fronted IP configurations.

  For:

  - public IPs  - these are zones in which the public IP resource is available.
  - private IPs - these are zones to which Azure will deploy paths leading to Load Balancer frontend IPs (all frontends are 
                  affected).

  Setting this variable to explicit `null` disables a zonal deployment.
  This can be helpful in regions where Availability Zones are not available.

  For public Load Balancers, since this setting controls also Availability Zones for public IPs, you need to specify all zones
  available in a region (typically 3): `["1","2","3"]`.
  EOF
  default     = ["1", "2", "3"]

  nullable = true  #<-- flip this value to true instead

  type     = list(string)
}
```  

### Current behavior

Setting `zones` variable to `null` will result in the variable value being set to the default value of `["1", "2", "3"]` which prevents being able to disable zonal deployment. I also attempted to set this value to `[]` (empty list) and it failed to deploy due to bad value.

**Comment by acelebanski** [comment]
Hi @jinkang23, another good catch! This will be fixed in PR #62, thank you!

**Comment by jinkang23** [comment]
Thank you, @acelebanski
